reframe/core/modules.py

```diff
@@ -203,6 +204,32 @@ def is_module_loaded(self, name):
     def _is_module_loaded(self, name):
         return self._backend.is_module_loaded(Module(name))
 
+    def load_mapping(self, mapping):
+        """Updates the internal module mapping with a single mapping"""
+        key, *rest = mapping.split(':')
+        if len(rest) != 1:
+            raise ConfigError('invalid mapping syntax: %s' % mapping)
+
+        key = key.strip()
+        values = rest[0].split()
+        if not key or not values:
+            raise ConfigError('invalid mapping syntax: %s' % mapping)
+
+        self.module_map[key] = list(OrderedDict.fromkeys(values))
+
+    def load_mapping_from_file(self, filename):
+        """Update the internal module mapping from mappings in a file."""
+        with open(filename) as fp:
+            for lineno, line in enumerate(fp):
```

Comment:
You should pass `start=1` to `enumerate` so as to count the lines in a human readable way.